### ISSUE TITLE
`floor(pi)` should work

### ISSUE THREAD

**Issue by eschnett** [JuliaLang/julia#17080]
See https://www.reddit.com/r/Julia/comments/4pjr2m/getting_the_first_digit_of_a_mathematical_constant/.

`floor(pi)` currently returns

```
julia> floor(pi)
ERROR: MethodError: no method matching floor(::Irrational{:π})
```

It should probably calculate `floor(float(pi))` instead.

**Comment by simonster** [comment]
Very similar to #5561

**Comment by ararslan** [comment]
An idea that may or may not be horrible would be to define `Irrational <: AbstractFloat` rather than `<: Real`. I guess it's accurate, and it permits more methods with minimal changes, but it kind of defeats the purpose of the `Irrational` type...

**Comment by StefanKarpinski** [comment]
No, that's straight up incorrect: `Irrational` and `AbstractFloat` are disjoint concepts – floats can *only* represent rational values and huge amounts of code rely on this fact and other code relies on the fact that no irrational can ever equal any rational and thus any float.

**Comment by ararslan** [comment]
I guess I meant more in an abstract, mathematical sense rather than from an actually useful, practical perspective. Like how given infinite decimal places you could exactly represent an irrational value. But you're right, that doesn't translate to anything useful in practice.

**Comment by mbauman** [comment]
`floor(pi)` does now work!

```
julia> floor(pi)
3.0

julia> ceil(pi)
4.0

julia> round(pi)
3.0
```

**Comment by waldyrious** [comment]
> `floor(pi)` does now work!

For the record, this was fixed by #26760.

Also, noting that `round(pi)` is now covered in https://github.com/JuliaLang/julia/blob/8c70b0a624d55f960d91d1fb448093614fa76c48/test/rounding.jl#L212-L213, would it make sense to also add tests for `ceil(pi)` and `floor(pi)` under https://github.com/JuliaLang/julia/blob/8c70b0a624d55f960d91d1fb448093614fa76c48/test/rounding.jl#L234?
